src/smoothscroll.js

```diff
@@ -93,22 +93,37 @@
      */
     function findScrollableParent(el) {
       var isBody;
-      var hasScrollableSpace;
-      var hasVisibleOverflow;
+      var hasVerticalScrollableSpace;
+      var hasHorizontalScrollableSpace;
+      var hasScrollableOverflowY;
+      var hasScrollableOverflowX;
+      var isVerticallyScrollable;
+      var isHorizontallyScrollable;
 
       do {
         el = el.parentNode;
 
         // set condition variables
         isBody = el === d.body;
-        hasScrollableSpace =
-          el.clientHeight < el.scrollHeight ||
-          el.clientWidth < el.scrollWidth;
-        hasVisibleOverflow =
-          w.getComputedStyle(el, null).overflow === 'visible';
-      } while (!isBody && !(hasScrollableSpace && !hasVisibleOverflow));
 
-      isBody = hasScrollableSpace = hasVisibleOverflow = null;
+        hasVerticalScrollableSpace = el.clientHeight < el.scrollHeight;
+        hasHorizontalScrollableSpace = el.clientWidth < el.scrollWidth;
+        hasScrollableOverflowY =
+          w.getComputedStyle(el, null).overflowY === 'auto' ||
+          w.getComputedStyle(el, null).overflowY === 'scroll';
+        hasScrollableOverflowX =
+          w.getComputedStyle(el, null).overflowX === 'auto' ||
+          w.getComputedStyle(el, null).overflowX === 'scroll';
```

Comment:
Instead of repeating this logic I think is better to create a method which receives a node and returns true or false to avoid duplicated code.